.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,7 @@
+- id: verilator
+  name: verilator-lint
+  description: Runs verilator Docker image to lint (System) Verilog designs
+  args: [--lint-only, -Wall]
+  language: docker_image
+  entry: verilator/verilator:v5.026
```

Comment:
It's good practice to pin the version and not just point to latest. Is this acceptable, or should we pull the version from somewhere else (not sure if possible in yaml, though)?

Reply:
When is this file read by docker?  Will ./configure already have happened?  If so you can rename this to .pre-commit-hooks.yaml.in and have autoconf fill in the version number.  If not we can edit each release, though this is a bit error prone - I probably should automate what's currently a manual process.

Reply:
In any repo that uses this hook via their pre-commit-config

```yaml
  - repo: https://github.com/verilator/verilator
    rev: v5.026
    hooks:
      - id: verilator
```

pre-commit will just call

1.  `git clone` on this repo, in our case just to get `.pre-commit-hooks.yaml`
2. `docker run`, which automatically fetches the verilator docker image mentioned in `.pre-commit-hooks.yaml`

So this won't call ./configure at all. I don't see where we can intercept these steps to inject the version number. I see 3 options going forward:

1. As you suggest, automate the release process
2. As [hadolint actually does](https://github.com/hadolint/hadolint/blob/742b76e3b9b4d45d6716771b0f1fae2fa0069104/.pre-commit-hooks.yaml#L7), don't mention the version number in our pre-commit-hooks.yaml, but go for latest. I'm just seeing that the [pre-commit docs also mention `latest`](https://pre-commit.com/#docker_image), so maybe that's not a bad practice after all.
3. Make this a python pre-commit hook that calls docker under the hood. Then we can insert our version getting logic in the python hook. This isn't great because it makes dependency to docker less obvious to users.

I'd suggest going for option 2 for this hook, WDYT?

Btw, should I also document usage of this hook somewhere? Maybe in the toplevel readme?

Reply:
2 seems good, but shouldn't it be "stable" instead as a tag?  We don't want to rebuild on every commit...

Reply:
You don't build a new docker image on every commit, right? Only on tag creation, if I see that correctly. Also, the [stable verilator docker image is 3 years old at this time](https://hub.docker.com/r/verilator/verilator/tags), I think we want to be more recent than that.

With these aspect in mind, I suggest going for `verilator/verilator:latest`.

Reply:
Yes, only on tags.  We are a Docker approved open source team project, so we get 400 build minutes per month.  We have ~100 builds/month and each build is 5-10 minutes, so we don't have enough minutes to build every commit.

Yes, it's out of date, which is why I asked if you're willing to get it working and keep up support in #5238, as there hasn't been much interest/complaint about it being old.

Reply:
I see. I'm willing to support with providing a working docker image when we run into issues.

With that, would you go for `stable` or `latest`?

Reply:
Great.  Stable, since by my math above it's too much build time to use latest.

Reply:
I think we have a misunderstanding: `docker_image:latest` points to the latest tag of a docker image, not to the head of verilator's github master branch. So, if we use `latest`, we don't force ourselves to do additional docker builds.

Do you have other reasons for preferring `stable` over `latest`?

Reply:
Latest is ok then, that equates to Verilator's "stable" ;)

Reply:
Great!

```suggestion
  entry: verilator/verilator:latest
```